.github/workflows/builder_node_slsa3.yml

```diff
@@ -0,0 +1,388 @@
+  #                 Generate the SLSA provenance                    #
+  #                                                                 #
+  ###################################################################
+  provenance:
+    runs-on: ubuntu-latest
+    needs: [builder, build, rng]
+    permissions:
+      id-token: write # Needed to create an OIDC token for keyless signing.
+      contents: read
+      actions: read # Needed to read workflow info.
+    outputs:
+      node-provenance-name: ${{ steps.sign-prov.outputs.signed-provenance-name }}
+      node-provenance-sha256: ${{ steps.sign-prov.outputs.signed-provenance-sha256 }}
+    steps:
+      - name: Download builder
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-artifact@e3220805577deb9d193f64e519abcb3b50851df5
+        with:
+          name: "${{ env.BUILDER_BINARY }}-${{ needs.rng.outputs.value }}"
+          path: "${{ env.BUILDER_BINARY }}"
+          sha256: "${{ needs.builder.outputs.node-builder-sha256 }}"
+          set-executable: true
+
+      - name: Create and sign provenance
+        id: sign-prov
+        env:
+          UNTRUSTED_TARBALL_NAME: "${{ needs.build.outputs.node-tarball-name }}"
+          UNTRUSTED_TARBALL_HASH: "${{ needs.build.outputs.node-tarball-sha256 }}"
+          UNTRUSTED_CI_ARGUMENTS: "${{ inputs.ci-arguments }}"
+          UNTRUSTED_RUN_SCRIPTS: "${{ inputs.run-scripts }}"
+          UNTRUSTED_DIR: ${{ inputs.directory }}"
+          GITHUB_CONTEXT: "${{ toJSON(github) }}"
+        run: |
+          set -euo pipefail
+
+          echo "provenance generator is $BUILDER_BINARY"
+
+          # Create and sign provenance
+          # This sets signed-provenance-name to the name of the signed DSSE envelope.
+          # Note: this command outputs the sha256 of the signed provenance, so the upload
+          # in the next step need not be secure-artifact-upload.
+          ./"$BUILDER_BINARY" provenance \
+            --tarball-name "$UNTRUSTED_TARBALL_NAME" \
+            --digest "$UNTRUSTED_TARBALL_HASH" \
+            --ci-arguments "$UNTRUSTED_CI_ARGUMENTS" \
+            --run-scripts "$UNTRUSTED_RUN_SCRIPTS" \
+            --directory "$UNTRUSTED_DIR"
+
+      - name: Upload the signed provenance
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # tag=v3.1.0
+        with:
+          name: "${{ steps.sign-prov.outputs.signed-provenance-name }}"
+          path: "${{ steps.sign-prov.outputs.signed-provenance-name }}"
+          if-no-files-found: error
+          retention-days: 5
+
+  ###################################################################
+  #                                                                 #
+  #           Publish the package and its provenance                #
+  #                                                                 #
+  ###################################################################
+  publish:
```

Comment:
I'm guessing we should probably have an option that allows you to skip the publishing step.

Reply:
How about we don't provide it right away and see if people ask for it? By default, this option would be set to `true`, so we could add it later. Wdut?